general/mm.hpp

```diff
@@ -70,21 +88,32 @@ class mm
    // * Allocates n*size bytes and returns a pointer to the allocated memory
    // **************************************************************************
    template<class T>
-   static inline T* malloc(const size_t n, const size_t size = sizeof(T))
-   { return (T*) MM().Insert(::new T[n], n*size); }
-
+   static inline T* malloc(const size_t n, 
+                           char const* const filename = __FILE__,
+                           int const lineno = __LINE__,
+                           char const* const function = __FUNCTION__,
+                           size_t const size = sizeof(T))
+   { return (T*) MM().Insert(::new T[n], n*size, filename, lineno, function); }
```

Comment:
I'm not a fan of `mm_malloc(T,n)`. Macros for something as simple as allocating memory using non-C++ syntax is a recipe for user confusion. I understand the benefit - that this gives us file and line numbers via the macros, but I think this can be solved with debuggers or `MFEM_USE_LIBUNWIND`.

I liked it better before this change... but what do others think?

Reply:
This is definitely temporary: I am building a tool that checks that all the `::new` and `::delete` are not playing with the registered `mm` memory. I had to switch for these macros, but you are right, there are other ways that should allow to do the same thing.

Reply:
👍